src/PkgAuthentication.jl

```diff
@@ -31,14 +31,17 @@ function _assert_pkg_server_env_var_is_set()
 end
 
 """
-    authenticate(server::AbstractString)
+    authenticate(server::AbstractString; kwargs...)
 
 Starts interactive (blocking) browser-based Pkg server authentication for the Pkg
 server specified by `server`. Also sets the `$(pkg_server_env_var_name)` environment
 variable to `server`.
 
 `server` must be the URL of a valid Pkg server.
 
+## Keyword arguments
+- `modify_environment::Bool = true`: Set the `$(pkg_server_env_var_name)` environment variable to `server`.
```

Comment:
```suggestion
- `modify_environment::Bool = true`: Set the `$(pkg_server_env_var_name)` environment variable to `server`. In package code, this should probably be set to `false`, so that the package would not have unexpected global side effects.
```